### PR TITLE
New Directive to populate sub field

### LINKED ISSUE #926: Api server - Users table requires sub column not part of CreateUserInput
# Description
We added a non-nullable 'sub' column to the users table. This column is to link a user in the api database to users/tokens from the authentication service.
The column is non-nullable, but isn't set in the CreateUser mutation. This causes the CreateUser mutation to error.
When creating a new User, the sub field should be set to match the email field. This should be handled in the Laravel model, not the schema.

### PR DESCRIPTION
**Pull request by petertgiles**
A new directive added to populate the sub field directly, not as part of the graphql mutation.  I couldn't find a built-in directive to do this so I created a new one based on the inject directive.

Closes #926

### CODE REVIEW

File: api/graphql/schema.graphql [hunk 1]
```diff
@@ -439,7 +439,7 @@ input UpdatePoolCandidateSearchRequestInput {
 }
 
 type Mutation {
-    createUser(user: CreateUserInput! @spread): User @create @guard @can(ability: "create")
+    createUser(user: CreateUserInput! @spread): User @create @guard @can(ability: "create") @populateSub(source: "email")
```
Comment (petertgiles) [inline api/graphql/schema.graphql:442]: Would it make more sense to have a generic "copy field a to b" directive?  Or is "sub" going to be a standardized column that so it is useful to have its own directive?

Comment (petertgiles) [reply api/graphql/schema.graphql:442]: Or even less generic?  If it will always be email to sub then make the directive do that with no arguments?

Comment (tristan-orourke) [reply api/graphql/schema.graphql:442]: I really like the idea of the more generic, copy a to b, version

**Review by tristan-orourke — CHANGES_REQUESTED** [review]
Nice work figuring out a custom directive. I really like your proposal of a more generic version that can copy any one field to another, lets go with that.

File: api/graphql/schema.graphql (hunk 1 above)
Comment (petertgiles) [reply api/graphql/schema.graphql]: Committed a new version that can copy from a to b

**Review by tristan-orourke — APPROVED** [review]

**Pull request merged**